WcaOnRails/lib/time_limit.rb

```diff
@@ -1,16 +1,29 @@
 # frozen_string_literal: true
 
 class TimeLimit
+  include ActiveModel::Validations
+
   TEN_MINUTES_IN_CENTISECONDS = 10*60*100
 
-  attr_accessor :centiseconds, :cumulative_round_ids
-  def initialize(centiseconds: TEN_MINUTES_IN_CENTISECONDS, cumulative_round_ids: [].freeze)
+  attr_accessor :centiseconds, :cumulativeRoundIds
+  validates :centiseconds, numericality: { only_integer: true }
+  validate do
+    unless self.cumulativeRoundIds.is_a?(Array) && self.cumulativeRoundIds.all? { |el| el.is_a?(String) }
+      errors.add("cumulativeRoundIds must be an Array of Strings")
+    end
+  end
+
+  def initialize(centiseconds: TEN_MINUTES_IN_CENTISECONDS, cumulativeRoundIds: [].freeze)
     self.centiseconds = centiseconds
-    self.cumulative_round_ids = cumulative_round_ids
+    self.cumulativeRoundIds = cumulativeRoundIds
+  end
+
+  def cumulativeRoundIds=(cumulativeRoundIds) # rubocop:disable Style/MethodName
```

Comment:
Line 41 isn't the only place where we set `cumulativeRoundId`s, there's also our constructor. This is the one place to handle both of those.

I've changed this to an `attr_reader` to hopefully make it less weird.